core/editor/src/misc/test_utils.rs

```diff
@@ -19,9 +19,8 @@ pub trait EditorTestUtils {
 	fn drag_tool(&mut self, typ: ToolType, x1: u32, y1: u32, x2: u32, y2: u32);
 	fn move_mouse(&mut self, x: u32, y: u32);
 	fn mousedown(&mut self, state: MouseState);
-	fn mouseup(&mut self, state: MouseState);
+	fn mouseup(&mut self, x: u32, y: u32);
```

Comment:
Question: is the inconsistency between `x: u32, y: u32` for `mouseup` and `state: MouseState` for `mousedown` intentional?

Reply:
There might be an issue with how I named the functions but my idea was that a for `mousedown` one wants to provide which `MouseKeys` are pressed while `mouseup` releases all `MouseKeys`. I did this because if we have separate `left_mouseup`and `right_mouseup`and `center_mouseup` functions we'd need to keep track of which mouse buttons are currently pressed to be able to only realase the correct buttons.

Reply:
I see. But what about, say, if both LMB and RMB are down but then only RMB is released (mouseup)?

Also, the rest of the code tends to use the terminology LMB, RMB, and MMB instead of Left, Right, Middle. This is just for tests, but it's up to you if you want to rename it accordingly or keep it as is.

Reply:
Do you think it would be better to have `fn mouseup(&mut self, state: MouseState);` to make this more clear and consistent?

Reply:
It would make it more consistent, but I'll let you decide if it would make it more clear. I don't require a change, but I bring it up as a potentially confusing/inconsistent part of the code.